ladder/teams/README.md

```diff
@@ -0,0 +1,51 @@
+# How Cilium Org Teams Are Managed
+Each Cilium org team that wishes to self-manage its team membership can do so using this directory.
```

Comment:
What exactly is a Cilium org team?

Reply:
githubism 
I'll look at the ladder language and see if i can normalize

Reply:
I added some more text at the top to disambiguate